src/client_side.cc

```diff
@@ -3358,18 +3354,19 @@ clientHttpConnectionsOpen(void)
         }
 
 #if USE_OPENSSL
-        if (s->flags.tunnelSslBumping) {
+        if (s->flags.tunnelSslBumping()) {
+            auto &rawFlags = s->flags.rawConfig();
             if (!Config.accessList.ssl_bump) {
                 debugs(33, DBG_IMPORTANT, "WARNING: No ssl_bump configured. Disabling ssl-bump on " << scheme << "_port " << s->s);
-                s->flags.tunnelSslBumping = false;
+                rawFlags.tunnelSslBumping = false;
             }
             if (!s->secure.staticContext && !s->secure.generateHostCertificates) {
                 debugs(1, DBG_IMPORTANT, "Will not bump SSL at " << scheme << "_port " << s->s << " due to TLS initialization failure.");
-                s->flags.tunnelSslBumping = false;
+                rawFlags.tunnelSslBumping = false;
```

Comment:
Please move this configuration adjustments to the configuration post-parsing adjustment code in configDoConfigure(). The idea behind these adjustments is flawed. It has created at least one subtle bug already AFAICT, but it is a difficult problem to fix, and _this_ unrelated PR should not try to fix it. Here, I just want to make sure that all flags.rawConfig() uses are confined to configuration code (including post-parsing adjustments in configDoConfigure()). Once those port configuration flags are finalized, they should not be modified.

Reply:
Done (72e6808).